flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerBase.java

```diff
@@ -835,8 +837,15 @@ public void updateAccumulators(final AccumulatorSnapshot accumulatorSnapshot) {
                         mainThreadExecutor);
     }
 
-    private void startCheckpointScheduler(final CheckpointCoordinator checkpointCoordinator) {
+    @Override
```

Comment:
The `CheckpointCoordinator` does not change during the lifetime of an `ExecutionGraph`. That's why, I thought of making this change to introduce the `CheckpointScheduling` interface.